cli/src/main/java/org/eclipse/hono/cli/AmqpSend.java

```diff
@@ -89,7 +101,19 @@ void start() {
             return senderTracker;
         })
         .map(sender -> {
-            sender.send(message, delivery -> messageSent.complete(delivery));
+            if (subscribe) {
+                subscribeToCommands((d, m) -> {
+                    LOG.info("Received Command Message : {}", m.getSubject());
+                    final Message commandResponse = ProtonHelper.message(m.getReplyTo(), "OK: " + m.getSubject());
+                    commandResponse.setCorrelationId(m.getCorrelationId());
+                    MessageHelper.addProperty(commandResponse, MessageHelper.APP_PROPERTY_STATUS, HttpURLConnection.HTTP_OK);
+                    commandResponse.setContentType(m.getContentType());
+                    sender.send(commandResponse, delivery -> messageSent.complete(delivery));
+                });
+            } else {
+                final Message message = ProtonHelper.message(messageAddress, payload);
+                sender.send(message, delivery -> messageSent.complete(delivery));
+            }
```

Comment:
I think it would improve readability if we could factor out this whole block into a separate method that accepts a `ProtonSender`, checks the *subscribe* flag, creates the command consumer and returns a `Future<ProtonSender>`, don't you think? That way we could also add decent error handling for the case that creating the consumer doesn't work ...

Reply:
yes, that make sense!